src/main/java/de/zalando/aruha/nakadi/config/JsonConfig.java

```diff
@@ -33,14 +39,15 @@
     @Bean
     @Primary
     public ObjectMapper jacksonObjectMapper() {
-        final ObjectMapper objectMapper = new ObjectMapper()
-                .setPropertyNamingStrategy(CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES);
+        final ObjectMapper objectMapper = new ObjectMapper();
 
+        objectMapper.setPropertyNamingStrategy(CAMEL_CASE_TO_LOWER_CASE_WITH_UNDERSCORES);
         objectMapper.registerModule(enumModule());
+        objectMapper.registerModule(eventTypeSchemaModule());
```

Comment:
This is a very convenient idea if the followin holds:
- API spec is changed to support schema as object in the case type is json_schema, still be a blob of text otherwise. 
- migration script exists and is applied to all existing event types.